nixos/modules/services/misc/calibre-server.nix

```diff
@@ -0,0 +1,77 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.calibre-server;
+
+in
+
+{
+
+  ###### interface
+
+  options = {
+
+    services.calibre-server = {
+
+      enable = mkEnableOption "calibre-server";
+
+      libraryDir = mkOption {
+        default = "/tmp/calibre-server";
+        description = ''
+          The directory where the Calibre library to serve is.
+          '';
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = "calibre-server";
+        description = "User account under which calibre-server runs.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "calibre-server";
+        description = "Group account under which calibre-server runs.";
+      };
+
+    };
+
+  };
+
+
+  ###### implementation
+
+  config = mkIf cfg.enable {
+
+    systemd.services.calibre-server =
+      {
+        description = "calibre-server, an OPDS server for a Calibre library";
```

Comment:
Systemd descriptions should be short and capitalized because they are shown in boot messages (e.g. "Starting Calibre Server...").